CreeDictionary/tests/test_docs.py

```diff
@@ -0,0 +1,148 @@
+from pathlib import Path
+from gitignore_parser import parse_gitignore
+from os import path
+import os
+from typing import List, Callable, NoReturn
+import toml
+import pytest
+from sys import stderr
+from functools import partial
+from sortedcontainers import SortedSet
+
+IgnoreCheck = Callable[[Path], bool]
+"""
+If a path is ignored, it gives True. If a path is not ignored, it gives False
+"""
+
+FILE_DIR = Path(path.dirname(__file__))
+
+
+def get_project_root():
+    """
+    bubbles up until we see a directory with .git folder
+
+    :return: The root directory of the repository
+    """
+    # Uhh I actually don't know any reliable way to do this
+    # this sounds like a reasonable method. You are Welcome to fix it.
```

Comment:
This is a tough problem in general, and I think this might be the best way to solve it :/

Of course, this is b0rked if we distribute the code outside of a repo, which is likely to happen in a Docker deployment.

Of course, I don't think this test will run OUTSIDE of a git repo so :S